pkg/recipes/terraform/config/config_test.go

```diff
@@ -647,8 +647,15 @@ func Test_AddProviders(t *testing.T) {
 			err = json.Unmarshal(expectedConfigBytes, &expectedConfig)
 			require.NoError(t, err)
 
-			// This performs a deep comparison of the two maps.
-			require.Equal(t, expectedConfig, actualConfig)
+			if tc.desc == "valid all supported providers" {
```

Comment:
I think instead of checking for a specific test, we should check the properties one by one in the actual and expected maps. @sk593's suggestion is also a good one.

Reply:
We cannot match values/properties of the providers for reasons i have commented. There are other unit tests that specifically test ucpProvider BuildConfig() under radius-project/radius/pkg/recipes/terraform/config/providers/kubernetes_test.go /azure_test.go/aws_test.go.